cmd/common/image/build.go

```diff
@@ -217,6 +200,138 @@ func runBuild(opts *cmdOpts) error {
 	return nil
 }
 
+func buildLocal(opts *cmdOpts, conf *buildFile, output string) error {
+	makefilePath := filepath.Join(output, "Makefile")
+	if err := os.WriteFile(makefilePath, makefile, 0o644); err != nil {
+		return fmt.Errorf("writing Makefile: %w", err)
+	}
+
+	buildCmd := exec.Command(
+		"make", "-f", makefilePath, "-j", "2",
+		"EBPFSOURCE="+conf.EBPFSource,
+		"OUTPUTDIR="+output,
+		"CFLAGS="+conf.CFlags,
+	)
+	if out, err := buildCmd.CombinedOutput(); err != nil {
+		return fmt.Errorf("build script: %w: %s", err, out)
+	}
+
+	return nil
+}
+
+func buildInContainer(opts *cmdOpts, conf *buildFile, output string) error {
+	cwd, err := os.Getwd()
+	if err != nil {
+		return fmt.Errorf("getting current directory: %w", err)
+	}
+
+	ctx := context.Background()
+	cli, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+	if err != nil {
+		return fmt.Errorf("creating docker client: %w", err)
+	}
+	defer cli.Close()
+
+	f := filters.NewArgs()
+	f.Add("reference", opts.builderImage)
+
+	images, err := cli.ImageList(ctx, types.ImageListOptions{Filters: f})
+	if err != nil {
+		return fmt.Errorf("listing images: %w", err)
+	}
+
+	var found bool
+
+	for _, img := range images {
+		for _, tag := range img.RepoTags {
+			if tag == opts.builderImage {
+				found = true
+				break
+			}
+		}
+
+		if found {
+			break
+		}
+	}
+
+	if !found {
+		fmt.Printf("Pulling builder image %s\n", opts.builderImage)
+		reader, err := cli.ImagePull(ctx, opts.builderImage, types.ImagePullOptions{})
+		if err != nil {
+			return fmt.Errorf("pulling builder image: %w", err)
+		}
+		io.Copy(io.Discard, reader)
+		defer reader.Close()
+	}
+
+	resp, err := cli.ContainerCreate(
```

Comment:
It works without that. Probably because it's all being run as root. I added the user to make it more reliable if we allow to run is as non sudo later on.